### DISCUSSION TITLE
Socket.IO cannot establish a connection

### DISCUSSION THREAD

**Discussion by marmon179** [socketio/socket.io#4365] [Q&A]
I use socket.io-client for Javascript(4.5.0) server on JAVA and when I enter address 'ws://10.201.223.67:9902/' the request is not by 'ws' but by 'http'.
<img width="797" alt="Screenshot_1" src="https://user-images.githubusercontent.com/74657342/167237224-0e8b560c-4543-4ab8-83db-c658aa4ca11b.png">
It returns

<img width="889" alt="Screenshot_2" src="https://user-images.githubusercontent.com/74657342/167237237-ada0eacc-083c-455a-a750-877ba92867dc.png">
Although I understand it should just come back 0
<img width="269" alt="Screenshot_3" src="https://user-images.githubusercontent.com/74657342/167237247-de9849c2-c385-43b5-bf4e-df4a192acda8.png">
It keeps trying to connect.

<img width="655" alt="Screenshot_4" src="https://user-images.githubusercontent.com/74657342/167237305-e2c18c50-300c-477e-8bd5-09827874cff2.png">

**Comment by darrachequesne** [comment]
Which server implementation are you using? Could you please check if it's implementing the latest Socket.IO protocol?

See also: https://socket.io/docs/v4/troubleshooting-connection-issues/#problem-the-socket-is-not-able-to-connect

**Comment by marmon179** [comment]
###      Hello.client socket.io (2.4.0 JavaScript), I want to connect to the Socket.IO JAVA server (2.x.x).When I set transports: ['websocket', 'polling']. The request doesn't go at all.When I request through Postman the connection is established.

<img width="843" alt="Screenshot_6" src="https://user-images.githubusercontent.com/74657342/167362529-88f8eafe-9f72-4fdd-946c-efb0e9386da7.png">

### My code 

      `    export function useSocketIO(url: string) {
            const ioRef = useRef<Socket>();
          const [connected, setConnected] = useState(false);

        useEffect(() => {
         ioRef.current = io(url, {
        transports: ['websocket', 'polling'],
        });
         ioRef.current.on('connect', () => {
          console.log('WS Connected to', url);
           setConnected(() => true);
         });

    ioRef.current.on('disconnect', () => {
      console.log('WS disconnected');
      setConnected(() => false);
    });

           return () => {
           console.log('WS destroyed');
             ioRef.current?.close();
           };
           }, [url]);

           return {
            io: ioRef.current,
           connected,
          };
         }

           export type RealtimeOptions<T> = {
           event: string;
           onEvent?: (data: T) => void;
          };

           export type CollectionService<T> = {
           options: RealtimeOptions<T>;
          events: Array<T>;
          send: (data: T) => void;
          emit: (event: string, data: T) => void;
         };

       
         const DEFAULT_URL = 'ws://10.201.223.67:9902/';

       export function useRealtime<T>(options: RealtimeOptions<T>): CollectionService<T> {
        const [url] = useState(DEFAULT_URL);
        const { io } = useSocketIO(url);
        const [events, setEvents] = useState<Array<T>>([]);

          useEffect(() => {
           io?.on(options.event, (data: T) => {
            setEvents((events) => {
             return [...events, data];
            });
           options.onEvent?.(data);
           });
          }, [io, options, events, options.event]);

         return {
          options,
          events,
           send: (data) => io?.send(data),
         emit: (event, data) => io?.emit(event, data),
         };
     }`
